test/model/message_list_test.dart

```diff
@@ -371,70 +372,92 @@ void main() async {
         checkNotNotified();
-        check(model).messages.single.reactions.jsonEquals([]);
+        check(reactionsDotAddCalls).isEmpty();
       });
 
       test('remove reaction', () async {
-        final eventReaction = Reaction(reactionType: ReactionType.unicodeEmoji,
-          emojiName: 'wave',          emojiCode: '1f44b', userId: 1);
-
-        // Same emoji, different user. Not to be removed.
-        final reaction2 = Reaction(reactionType: ReactionType.unicodeEmoji,
-          emojiName: 'wave',          emojiCode: '1f44b', userId: 2);
-
-        // Same user, different emoji. Not to be removed.
-        final reaction3 = Reaction(reactionType: ReactionType.unicodeEmoji,
-          emojiName: 'working_on_it', emojiCode: '1f6e0', userId: 1);
-
-        // Same user, same emojiCode, different emojiName. To be removed: servers
-        // key on user, message, reaction type, and emoji code, but not emoji name.
-        // So we mimic that behavior; see discussion:
-        //   https://github.com/zulip/zulip-flutter/pull/256#discussion_r1284865099
-        final reaction4 = Reaction(reactionType: ReactionType.unicodeEmoji,
-          emojiName: 'hello',         emojiCode: '1f44b', userId: 1);
-
-        final originalMessage = eg.streamMessage(
-          reactions: [reaction2, reaction3, reaction4]);
+        final originalMessage = eg.streamMessage(reactions: Reactions([]));
         prepare();
         await prepareMessages(foundOldest: true, messages: [originalMessage]);
         final message = model.messages.single;
 
+        final List<({ReactionType reactionType, String emojiCode, int userId})> reactionsDotRemoveCalls = [];
+        message.reactions!.debugOnRemove = ({
+          required ReactionType reactionType,
+          required String emojiCode,
+          required int userId,
+        }) {
+          reactionsDotRemoveCalls.add((reactionType: reactionType, emojiCode: emojiCode, userId: userId));
+        };
+
+        final eventReaction = eg.unicodeEmojiReaction;
         model.maybeUpdateMessageReactions(
           mkEvent(eventReaction, ReactionOp.remove, originalMessage.id));
+
         checkNotifiedOnce();
-        check(model).messages.single
-          ..identicalTo(message)
-          ..reactions.jsonEquals([reaction2, reaction3]);
+        check(model).messages.single.identicalTo(message);
+        check(reactionsDotRemoveCalls).single
+          .which(it()
+            ..has((c) => c.reactionType, 'reactionType').equals(eventReaction.reactionType)
+            ..has((c) => c.emojiCode,       'emojiCode').equals(eventReaction.emojiCode)
+            ..has((c) => c.userId,             'userId').equals(eventReaction.userId),
+        );
       });
 
       test('remove reaction; message is not in list', () async {
-        final someMessage = eg.streamMessage(id: 1, reactions: [eg.unicodeEmojiReaction]);
+        final someMessage = eg.streamMessage(id: 1, reactions: Reactions([eg.unicodeEmojiReaction]));
```

Comment:
Seems like it's more convenient for callers of `eg.streamMessage` if this `reactions` parameter remains just a (nullable) list. The implementation can then say something like
```dart
  'reactions': reactions == null ? null : Reactions(reactions),
```